src/tests/src/libs/antares/study/thermal-price-definition/thermal-price-definition.cpp

```diff
@@ -0,0 +1,239 @@
+
+using namespace std;
+using namespace Antares::Data;
+
+std::string getFolder()
+{
+    std::filesystem::path tmpDir = std::filesystem::temp_directory_path();
+    return tmpDir.string();
+}
+
+void createIniFile()
+{
+    std::string folder = getFolder();
+
+    std::ofstream outfile;
+    outfile.open(folder + SEP + "list.ini", std::ofstream::out | std::ofstream::trunc);
+
+    outfile << "[area]" << std::endl;
+    outfile << "name = area" << std::endl;
+    outfile << "group = Lignite" << std::endl;
+    outfile << "unitcount = 4" << std::endl;
+    outfile << "nominalcapacity = 900.000000" << std::endl;
+    outfile << "min-stable-power = 400.000000" << std::endl;
+    outfile << "min-up-time = 24" << std::endl;
+    outfile << "min-down-time = 24" << std::endl;
+    outfile << "co2 = 1.200000" << std::endl;
+    outfile << "marginal-cost = 35.000000" << std::endl;
+    outfile << "fixed-cost = 1700.000000" << std::endl;
+    outfile << "startup-cost = 70000.000000" << std::endl;
+    outfile << "market-bid-cost = 35.000000" << std::endl;
+    outfile << "costgeneration = useCostTimeseries" << std::endl;
+    outfile << "efficiency = 35.350000" << std::endl;
+    outfile << "variableomcost = 12.120000" << std::endl;
+
+    outfile.close();
+}
+
+void createFuelCostFile(int size)
+{
+    std::string folder = getFolder();
+
+    std::ofstream outfile;
+    outfile.open(folder + SEP + "fuelCost.txt", std::ofstream::out | std::ofstream::trunc);
+
+    for (int i = 0; i < size; i++)
+    {
+        outfile << 1 << std::endl;
+    }
+}
+
+void createCo2CostFile(int size)
+{
+    std::string folder = getFolder();
+
+    std::ofstream outfile;
+    outfile.open(folder + SEP + "CO2Cost.txt", std::ofstream::out | std::ofstream::trunc);
+
+    for (int i = 0; i < size; i++)
+    {
+        outfile << 1 << std::endl;
+    }
+}
+
+void removeIniFile()
+{
+    std::string folder = getFolder();
+    std::filesystem::remove(folder + SEP + "list.ini");
+}
+
+void removeCostFiles()
+{
+    std::string folder = getFolder();
+    std::filesystem::remove(folder + SEP + "fuelCost.txt");
+    std::filesystem::remove(folder + SEP + "CO2Cost.txt");
+}
+
+// =================
+// The fixture
+// =================
+struct Fixture
+{
+    Fixture(const Fixture & f) = delete;
+    Fixture(const Fixture && f) = delete;
+    Fixture & operator= (const Fixture & f) = delete;
+    Fixture& operator= (const Fixture && f) = delete;
+    Fixture()
+    {
+        area = study->areaAdd("area");
+        study->parameters.include.thermal.minUPTime = true;
+        study->parameters.include.thermal.minStablePower = true;
+        study->parameters.include.reserve.spinning = true;
+
+    }
+    ~Fixture() = default;
+
+    std::string folder = getFolder();
+    ThermalClusterList clusterList;
+    Area* area;
+
+    Study::Ptr study = std::make_shared<Study>();
+
+};
+
+
+// ==================
+// Tests section
+// ==================
+
+BOOST_FIXTURE_TEST_SUITE(s, Fixture)
```

Comment:
General comment for test case name.
Try to make them explicit as in: what are we doing, what are we expecting.
Example: `EconomicInputData_loadFromFolder_too_small` => loading fail when not enough value in file